src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -528,50 +515,103 @@ private Type EmitExprListNode(AssociativeNode node)
-            EmitOpCode(OpCodes.Newarr, ot);
-            int elCount = -1;
-            foreach (var el in elements)
+            EmitArray(ot, eln.Exprs, (AssociativeNode el, int idx) => 
             {
-                EmitOpCode(OpCodes.Dup);
-                EmitOpCode(OpCodes.Ldc_I4, ++elCount);
-                var t = DfsTraverse(el);
+                Type t = DfsTraverse(el);
                 if (ot == typeof(object) && t.IsValueType)
                 {
                     EmitOpCode(OpCodes.Box, t);
                 }
-                if (ot == typeof(int))
-                {
-                    EmitOpCode(OpCodes.Stelem_I4);
-                }
-                else if (ot == typeof(long))
-                {
-                    EmitOpCode(OpCodes.Stelem_I8);
-                }
-                else if (ot == typeof(double))
+                if (ot == typeof(double))
                 {
                     if (t == typeof(int) || t == typeof(long))
                     {
                         EmitOpCode(OpCodes.Conv_R8);
                     }
-                    EmitOpCode(OpCodes.Stelem_R8);
                 }
-                else if (ot == typeof(bool))
+            });
+            return ot.MakeArrayType();
+        }
+
+        /// <summary>
+        /// Creates an array of type T on the evaluation stack.
+        /// This function uses boxing/unboxing to figure out the item opcode types. Performance might
+        /// be affected, so use this function for simple cases.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="items"></param>
+        private void EmitArray<T>(IEnumerable<T> items = null)
+        {
+            var t = typeof(T);
+            EmitArray(t, items, (T item, int idx) => 
+            {
+                object obj = item;
+                if(t == typeof(int))
                 {
-                    EmitOpCode(OpCodes.Stelem_I1);
+                    EmitOpCode(OpCodes.Ldc_I4, (int)obj);
                 }
-                else if (ot == typeof(char))
+                else if(t == typeof(long)) 
                 {
-                    EmitOpCode(OpCodes.Stelem_I2);
+                    EmitOpCode(OpCodes.Ldc_I8, (long)obj);
                 }
-                else
+                else if(t == typeof(double))
                 {
-                    EmitOpCode(OpCodes.Stelem_Ref);
+                    EmitOpCode(OpCodes.Ldc_R8, (double)obj);
                 }
+                else if(t == typeof(bool))
+                {
+                    EmitOpCode((bool)obj ? OpCodes.Ldc_I4_1 : OpCodes.Ldc_I4_0);
+                }
+                else if(t == typeof(char))
+                {
+                    EmitOpCode(OpCodes.Ldc_I4_S, (char)obj);
+                }
+                else if(t == typeof(string))
+                {
+                    EmitOpCode(OpCodes.Ldstr, (string)obj);
+                }
+            });
+        }
```

Comment:
Do you think this function is really useful? Right now, it replaces only 2 lines of code, lines 670 and 671 before this PR.

Reply:
Maybe not useful right now.
I can remove it for now and lines 670 and 671 could be replaced with a call to the other EmitArray function: 
EmitArray<string>(typeof(string), items: null, itemEmitter: null);